cosmocrat/helper_functions.py

```diff
@@ -0,0 +1,99 @@
+import os
+import re
+import sys
+import subprocess
+import cosmocrat.definitions as definitions
+
+from osmeterium.run_command import run_command
+from cosmocrat.definitions import log, Subprocess_Tool
+
+def log_and_exit(exception_message, exit_code):
+    log.error(exception_message)
+    sys.exit(exit_code)
+
+def subprocess_error_handler_wrapper(subprocess_name, already_raised_exit_code=None):
+    exit_code_value = 'general_error'
+    logged_message_head = 'The subprocess'
+    if subprocess_name in Subprocess_Tool._member_names_:
+        exit_code_value = f'{subprocess_name}_error'
+        logged_message_head = subprocess_name
+    def error_handler(exit_code=already_raised_exit_code):
+        log.error(fr'{logged_message_head} raised an error: {exit_code}')
+        sys.exit(definitions.EXIT_CODES[exit_code_value])
+    return error_handler
+
+def run_command_wrapper(command, subprocess_name=''):
+    run_command(command,
+                log.info,
+                log.info,
+                subprocess_error_handler_wrapper(subprocess_name),
+                (lambda: log.info('subprocess finished successfully.')))
+
+def subprocess_get_stdout_output(args, subprocess_name=''):
+    completed_process = subprocess.run(args=args,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        universal_newlines=True)
+    return_code = completed_process.returncode                   
+    if return_code is not 0:
+        subprocess_error_handler_wrapper(subprocess_name,
+                                        already_raised_exit_code=return_code)()
+    output = completed_process.stdout
+    return output.strip()
+
+def get_compression_method(compression, default_format=''):
+    compression_type = 'none'
+    compression_format = default_format
+    if compression:
+        compression_type = 'gzip'
+        compression_format += '.gz'
```

Comment:
you can use the values from `definitions.py`

Reply:
done.